core/chains/evm/types/models.go

```diff
@@ -168,11 +168,14 @@ func (h *Head) ChainHashes() []common.Hash {
 }
 
 func (h *Head) LatestFinalizedHead() commontypes.Head[common.Hash] {
-	for h != nil && !h.IsFinalized {
+	for h != nil {
+		if h.IsFinalized {
+			return h
+		}
 		h = h.Parent
 	}
 
-	return h
+	return nil
 }
```

Comment:
Was the logic changed here? Seems like it's doing the same thing as before

Reply:
It's a slight nuance, instead of returning a nil interface we explicitly return nil, see this discussion for more details: https://chainlink-core.slack.com/archives/C03ANJB9VB7/p1715163095224569?thread_ts=1714656712.938949&cid=C03ANJB9VB7